trending_data/src/main/webapp/unique_window_handler.js

```diff
@@ -0,0 +1,191 @@
+import {prepareYTPosts} from './handle_youtube.js';
+import {prepareTwitterPosts} from './handle_twitter.js';
+
+/**
+*  Class to keep only one open info window and
+* deal with displaying data. Saves data so that
+* it does not need to be re-fetched so long as
+* the window is open
+*/
+export class UniqueWindowHandler {
+  /**
+  * @param {Map} map
+  */
+  contructor(map) {
+    this.currentWindow = null;
+    this.map = map;
+    this.marker = marker;
+    this.countryCode = null;
+    this.countryName = null;
+  }
+
+  /**
+  * Initializes the divs where YouTube and Twitter data
+  * will be cached.
+  */
+  initDataDivs() {
+    this.ytDataDiv = document.createElement('div');
+    this.twitterDataDiv = document.createElement('div');
+  }
+
+  /**
+  * DataWindow contains a button container with the 2
+  * buttons (YouTube, Twitter) and the content corresponding
+  * to each platform. This method initializes the DataWindow
+  * with the button container.
+  */
+  initDataWindow() {
+    this.dataWindow = document.createElement('div');
+    this.dataWindow.className = 'popup';
+    this.dataWindow.appendChild(this.createBttnDiv());
+  }
+
+  /**
+  * Called when Twitter button is clicked. Clears
+  * current content of window and shows the most recently
+  * cached Twitter data
+  */
+  showYTData() {
+    this.initDataWindow(); // clear current content and re-add buttons
+    this.dataWindow.appendChild(this.ytDataDiv);
+    this.currentWindow.setContent(this.dataWindow);
+  }
+
+  /**
+  * Called when YouTube button is clicked. Clears
+  * current content of window and shows the most recently
+  * cached YouTube data
+  */
+  showTwitterData() {
+    this.initDataWindow();
+    this.dataWindow.appendChild(this.twitterDataDiv);
+    this.currentWindow.setContent(this.dataWindow);
+  }
+
+  /**
+  * Creates a container with 2 buttons, one for Twitter
+  * and one for YouTube and adds them to the top of the
+  * window where data will be displayed
+  * @return {HTMLElement} bttnDiv
+  */
+  createBttnDiv() {
```

Comment:
This method now contains different code in dev. It is unclear how the merging will occur since the modified code in dev is inside `script.js` which no longer exists in this PR